addons/website/models/website.py

```diff
@@ -241,11 +245,142 @@ def unlink(self):
+            custo = { 'color-palettes-name': "'%s'" % selected_palette, }
+            self.env['web_editor.assets'].make_scss_customization(url, custo)
+
+        def set_features(selected_features):
+            feature_ids = self.env['website.configurator.feature'].browse(selected_features)
+            pages_views = {}
+            modules = self.env['ir.module.module']
+            for feature_id in feature_ids:
+                if feature_id.type == 'app' and feature_id.module_id and feature_id.module_id.state != 'installed':
+                    modules += feature_id.module_id
+                if feature_id.type == 'page' and feature_id.page_view_id:
+                    result = self.env['website'].new_page(name=feature_id.name, add_menu=True, template=feature_id.page_view_id.key)
+                    pages_views[feature_id.iap_page_code] = result['view_id']
+
+            modules._button_immediate_function(lambda self:self.button_install())
+            return pages_views
+
+        def configure_page(page_code, snippet_list, pages_views):
+            if page_code == 'homepage':
+                page_view_id = website.homepage_id.view_id
+            else:
+                page_view_id = self.env['ir.ui.view'].browse(pages_views[page_code])
+            page_snippets = b''
+            for snippet in snippet_list:
+                try:
+                    view_id = self.env['website'].with_context(website_id=website.id).viewref(snippet)
+                    if view_id:
+                        page_snippets += view_id._render()
+                except ValueError as e:
+                    logger.warning(e)
+            page_view_id.save(value=page_snippets.decode('utf-8'), xpath="(//div[hasclass('oe_structure')])[last()]")
+
+        def set_images(images):
+            for image in images:
+                try:
+                    response = requests.get(image['url'],timeout=3)
+                    response.raise_for_status()
+                    datas = base64.b64encode(response.content).decode('utf-8')
+                    self.env['ir.attachment'].create({
+                        'name': image['name'],
+                        'website_id': website.id,
+                        'key': image['name'],
+                        'type': 'binary',
+                        'datas': datas,
+                    })
+                except requests.HTTPError:
+                    logger.warning("Failed to download image '%s'" % image['url'])
+
+        website = self.get_current_website()
+
+        url = self.env['ir.module.module'].search([('name', '=', data['themeName'])]).button_choose_theme()
+
+        website.configurator_done = True
+
+        # Enable tour
+        self.env['ir.asset'].create({
+            'name': 'website.configurator_tour',
+            'bundle': 'website.assets_editor',
+            'glob':'website/static/src/js/tours/configurator_tour.js',
+        })
```

Comment:
@Arcasias we want to enable this specific tour only when the user complete the website configurator. It works but is it the good way to do it ? @antonylesuisse

Reply:
I think a better way would be to declare it in an XML data file with its `active` field to false, and then activate it here (you can find an `ir.asset` record by name or URL)